File: extract-video-data/ocr.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

# index: 66e9358a808d95368f6f7a7c
# video: 66ebe4bfd4f452b062c63e3a

response = client.index.video.text_in_video(
    index_id="66e9358a808d95368f6f7a7c",
    id="66ebe4bfd4f452b062c63e3a",
    start=100,
    end=300
)

for i, text in enumerate(response, start=1):
    print(f"\nSegment {i}:")
    print(f"  Start time: {text.start} seconds")
    print(f"  End time: {text.end} seconds")
    print(f"  Text: {text.value}")

File: extract-video-data/ocr.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY});

  const response = await client.index.video.textInVideo(
    '66e9358a808d95368f6f7a7c',
    '66ebe4bfd4f452b062c63e3a',
    { start:100, end:300}  
)

  response.forEach((text, index) => {
    console.log(`\nSegment ${index + 1}:`);
    console.log(`  Start time: ${text.start} seconds`);
    console.log(`  End time: ${text.end} seconds`);
    console.log(`  Transcription: ${text.value}`);
  });
})();


